src/errors.js

```diff
@@ -178,6 +178,14 @@ class KafkaJSUnsupportedMagicByteInMessageSet extends KafkaJSNonRetriableError {
   }
 }
 
+class KafkaJSCorrelationIdAlreadyExists extends KafkaJSNonRetriableError {
+  constructor(correlationId) {
+    super(...arguments)
+    this.name = 'KafkaJSCorrelationIdAlreadyExists'
+    this.message = `Correlation id ${correlationId} already exists`
+  }
+}
```

Comment:
This error is a bit unusual in that it should _never_ happen. If it does happen, it likely means that there is a coding error somewhere. I would suggest we make this clear by creating a generic `KafkaJSInvariantViolation` and urge the user to file an issue:

```suggestion
const pkgJson = require('../package.json')
const issueUrl = pkjJson.bugs.url

class KafkaJSInvariantViolation extends KafkaJSNonRetriableError {
  constructor(e) {
    const message = e.message || e
    super(`Invariant violated: ${message}. This is likely a bug and should be reported.`)
    this.name = 'KafkaJSInvariantViolation'
    this.helpUrl = `${issueUrl}/new?assignees=&labels=bug&template=bug_report.md&title=${encodeURIComponent(`Invariant violation: ${message}`)}`
  }
}
```

```js
throw new KafkaJSInvariantViolation('Correlation id already exists')
```

Reply:
Thanks for your feedback, I changed the error as requested.